includes/Products/Sync/Background.php

```diff
@@ -139,9 +145,115 @@ public function process_items( $job, $data, $items_per_batch = null ) {
 	 * @param mixed $item
 	 * @param object|\stdClass $job
 	 * @return array
+	 * @throws Framework\SV_WC_Plugin_Exception
 	 */
-	public function process_item($item, $job) {
-		// TODO
+	public function process_item( $item, $job ) {
+
+		list( $product_id, $method ) = $item;
+
+		$product = wc_get_product( $product_id );
+
+		if ( ! $product instanceof \WC_Product ) {
+			throw new Framework\SV_WC_Plugin_Exception( "No product found with ID equal to {$product_id}." );
+		}
+
+		if ( ! in_array( $method, [ Sync::ACTION_UPDATE, Sync::ACTION_DELETE ], true ) ) {
+			throw new Framework\SV_WC_Plugin_Exception( "Invalid sync request method: {$method}." );
+		}
+
+		if ( Sync::ACTION_UPDATE === $method ) {
+			$request = $this->process_item_update( $product );
+		} else {
+			$request = $this->process_item_delete( $product );
+		}
+
+		return $request;
+	}
+
+
+	/**
+	 * Processes an UPDATE sync request for the given product.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @param \WC_Product $product product object
+	 */
+	private function process_item_update( $product ) {
+
+		$product_data = $this->prepare_product_data( $product );
+
+		return [
+			'retailer_id' => $product_data['retailer_id'],
+			'method'      => Sync::ACTION_UPDATE,
+			'data'        => $product_data,
+		];
+	}
+
+
+	/**
+	 * Prepares the product data to be included in a sync request.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @param \WC_Product $product product object
+	 * @return array
+	 * @throws Framework\SV_WC_Plugin_Exception
+	 */
+	private function prepare_product_data( $product ) {
+
+		if ( $product->is_type( 'variation' ) ) {
```

Comment:
Optional suggestion: extract the variation logic into its own method for clarity / cleaner code

Reply:
Sounds good. I added a couple of updates for that., starting from https://github.com/facebookincubator/facebook-for-woocommerce/pull/1320/commits/32b0e031a6de18f23b169359dbf88cf1e995f108. What do you think?

Reply:
Looks great! Thanks for changing that!